shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts

```diff
@@ -194,6 +200,20 @@ export class BaseCore implements IAppInsightsCore {
         return new ProcessTelemetryContext(thePlugins, _this.config, _this);
     }
 
+    /**
+     * Validate UUID Format
+     * Specs taken from https://tools.ietf.org/html/rfc4122
+     */
+    validateInstrumentationKey(iKey:string) :boolean {
```

Comment:
It also might be worth adding a "disableIkeyValidation" which defaults to false -- just in case we miss something.

Alternate suggestion: default to calling throwInternal() to log an error rather than throwing an exception -- that way we won't break anyone.

Reply:
Agree with Nev here. Seems like a good usecase to tie this in with the `enableDebug` config, which replaces all `throwInternal()` calls with a `throw`